### PR TITLE
8329099: Undocumented exception thrown by Instruction factory methods accepting Opcode

### PR DESCRIPTION
**Pull request by asotona**
`IllegalArgumentException` thrown by some static factory methods of the following `java.lang.classfile.instruction` interfaces are not documented:

- `ArrayLoadInstruction`
- `ArrayStoreInstruction`
- `BranchInstruction`
- `ConvertInstruction`
- `DiscontinuedInstruction`
- `FieldInstruction`
- `InvokeInstruction`
- `LoadInstruction`
- `MonitorInstruction`
- `NewPrimitiveArrayInstruction`
- `OperatorInstruction`
- `ReturnInstruction`
- `StackInstruction`
- `StoreInstruction`
- `TypeCheckInstruction`

`NewPrimitiveArrayInstruction::of` factory method also does not perform the check for invalid argument.

This patch adds all the missing `@throws` Javadoc tags and fixes `NewPrimitiveArrayInstruction::of` factory method to perform the argument check.

Please review.

Thank you,
Adam

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8329501](https://bugs.openjdk.org/browse/JDK-8329501) to be approved

### Issues
 * [JDK-8329099](https://bugs.openjdk.org/browse/JDK-8329099): Undocumented exception thrown by Instruction factory methods accepting Opcode (**Bug** - P4)
 * [JDK-8329501](https://bugs.openjdk.org/browse/JDK-8329501): Undocumented exception thrown by Instruction factory methods accepting Opcode (**CSR**)

### Reviewers
 * [Brian Goetz](https://openjdk.org/census#briangoetz) (@briangoetz - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18578/head:pull/18578` \
`$ git checkout pull/18578`

Update a local copy of the PR: \
`$ git checkout pull/18578` \
`$ git pull https://git.openjdk.org/jdk.git pull/18578/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18578`

View PR using the GUI difftool: \
`$ git pr show -t 18578`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18578.diff">https://git.openjdk.org/jdk/pull/18578.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18578#issuecomment-2031543176)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/classfile/instruction/NewPrimitiveArrayInstruction.java [hunk 1]
```diff
@@ -51,8 +52,13 @@ public sealed interface NewPrimitiveArrayInstruction extends Instruction
      * {@return a new primitive array instruction}
      *
      * @param typeKind the component type of the array
+     * @throws IllegalArgumentException when the {@code typeKind} is not a legal
+     *                                  primitive array component type
      */
     static NewPrimitiveArrayInstruction of(TypeKind typeKind) {
+        if (requireNonNull(typeKind).newarraycode() < 0) {
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/lang/classfile/instruction/NewPrimitiveArrayInstruction.java:59]:
```suggestion
        // Implicit null-check:
        if (typeKind.newarraycode() < 0) {
```

File: src/java.base/share/classes/java/lang/classfile/instruction/NewPrimitiveArrayInstruction.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 import java.lang.classfile.CodeModel;
 import java.lang.classfile.Instruction;
 import java.lang.classfile.TypeKind;
+import static java.util.Objects.requireNonNull;
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/lang/classfile/instruction/NewPrimitiveArrayInstruction.java:31]:
```suggestion
```

**Comment by ExE-Boss** [comment]
The following factory methods are also missing `@throws IllegalArgumentException`:
- `ConstantInstruction::ofIntrinsic`
- `ConstantInstruction::ofArgument`
- `ConstantInstruction::ofLoad`

**Comment by asotona** [comment]
> The following factory methods are also missing `@throws IllegalArgumentException`:
> 
> * `ConstantInstruction::ofIntrinsic`
> * `ConstantInstruction::ofArgument`
> * `ConstantInstruction::ofLoad`

Great catch, I'll add them, thank you!

**Review by briangoetz — APPROVED** [review]
While the @throws is not technically necessary here, specifying it here is reasonable.

**Pull request closed without merging**